testing/functional/tests/cypress/integration/student/02-login-page.spec.ts

```diff
@@ -0,0 +1,139 @@
+            LoginObject.cardNotFoundTxt()
+        })
+    })
+
+    it("Test with empty Card Serial Number such that it must get failed.", () => {
+        Welcome.virtualTestingBtn()
+        cy.fixture('testdata').then((testdata) => {
+            LoginObject.cardSerialNumberInputTxt()
+            LoginObject.cardSerialNumberContinuebtn()
+            LoginObject.emptyCardNumberTxt()
+        })
+    })
+
+    it("Verify that user can redirect to next page by pressing Enter key after typing in valid Card Serial Number.", () => {
+        Welcome.virtualTestingBtn()
+        cy.fixture('testdata').then((testdata) => {
+            LoginObject.cardSerialNumberInputTxt().type(testdata.cardSerialNumber).should('have.value', testdata.cardSerialNumber)
+            //LoginObject.cardSerialNumberContinuebtn()
+            LoginObject.enterKeyFromCardSerialNumber()
+            LoginObject.passcodePageTxt()
+        })
+    })
+
+    it("Verify that Card Serial Number with case sensitivity.", () => {
+        Welcome.virtualTestingBtn()
+        cy.fixture('testdata').then((testdata) => {
+            LoginObject.cardSerialNumberInputTxt().type(testdata.caseSenstivityCardSerialNumber)
+            LoginObject.caseSenstivityCardSerialNumber()
+
+        })
+    })
+
+    it("Verify that Card Serial Number must be within 15 digits allowed", () => {
+        Welcome.virtualTestingBtn()
+        cy.fixture('testdata').then((testdata) => {
+            LoginObject.cardSerialNumberInputTxt().type(testdata.fifteenDigitCardSerialNumber)
+            LoginObject.maxCharacterCardSerialNumber()
+
+        })
+    })
+
+    it.skip("Verify that user is not able to redirect to next page with valid Card Serial Number & invalid passcode.", () => {
+        Welcome.virtualTestingBtn()
+        cy.fixture('testdata').then((testdata) => {
+            LoginObject.cardSerialNumberInputTxt().type(testdata.validCardSerialNumber).should('have.value', testdata.validCardSerialNumber)
+            LoginObject.cardSerialNumberContinuebtn()
+            LoginObject.passcodeInputTxt().type(testdata.invalidPassocde).should('have.value', testdata.invalidPassocde)
+            LoginObject.passcodeContinurBtn()
+            LoginObject.incorrectPasscodeTxt()
+        })
+    })
+
+    it.skip("Test with empty passcode after entering valid card serial number such that it must get failed.", () => {
+        Welcome.virtualTestingBtn()
+        cy.fixture('testdata').then((testdata) => {
+            LoginObject.cardSerialNumberInputTxt().type(testdata.cardSerialNumber).should('have.value', testdata.cardSerialNumber)
+            LoginObject.cardSerialNumberContinuebtn()
+            LoginObject.passcodeInputTxt().type('  ')
+            LoginObject.passcodeContinurBtn()
+            LoginObject.whiteSpacePasscodeTxt()
+        })
+    })
+
+    it("Check a user is able to login by pressing Enter after entered valid Card Serial Number & valid passcode.", () => {
```

Comment:
After we click on the "sign-in"/"sign up" button the user is redirected to external entities (BCSC, BCeID, etc.) that will perform the user authentication. IMO we do not have to perform any sort of test on their system.

Reply:
What does it mean ? Are we not supposed to test this link ? But I have performed all login cases with positive and negative scenario.

Reply:
Everything that happens under the below UI is not in the scope of our solution and it is valid for BCeID, IDIR, and BCSC.
IMO these screens should be mapped to allow us to execute the authentication, but we do not need to run any type of test inside them.

For instance, the below test cases are outside of the scope of our tests, again IMO.

- Verify that user is not able to redirect to next page with invalid Card Serial Number in Virtual Card Testing.
- Test with empty Card Serial Number such that it must get failed.
- Verify that user can redirect to next page by pressing Enter key after typing in valid Card Serial Number.
- Verify that Card Serial Number with case sensitivity.
- Verify that Card Serial Number must be within 15 digits allowed
- Verify that user is not able to redirect to next page with valid Card Serial Number & invalid passcode.
- Test with empty passcode after entering valid card serial number such that it must get failed.
- and others...

![image](https://user-images.githubusercontent.com/61259237/159332709-dad6b2e7-3c6f-4eca-a5d4-ac7c2e863dce.png)